orderapi/src/main/java/fi/hel/verkkokauppa/order/service/accounting/AccountingSlipService.java

```diff
@@ -0,0 +1,201 @@
+package fi.hel.verkkokauppa.order.service.accounting;
+
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.common.error.Error;
+import fi.hel.verkkokauppa.common.util.DateTimeUtil;
+import fi.hel.verkkokauppa.common.util.UUIDGenerator;
+import fi.hel.verkkokauppa.order.api.data.accounting.AccountingSlipDto;
+import fi.hel.verkkokauppa.order.api.data.accounting.AccountingSlipRowDto;
+import fi.hel.verkkokauppa.order.api.data.accounting.OrderItemAccountingDto;
+import fi.hel.verkkokauppa.order.api.data.transformer.AccountingSlipRowTransformer;
+import fi.hel.verkkokauppa.order.api.data.transformer.AccountingSlipTransformer;
+import fi.hel.verkkokauppa.order.api.data.transformer.OrderItemAccountingTransformer;
+import fi.hel.verkkokauppa.order.model.Order;
+import fi.hel.verkkokauppa.order.model.accounting.AccountingSlip;
+import fi.hel.verkkokauppa.order.model.accounting.AccountingSlipRow;
+import fi.hel.verkkokauppa.order.model.accounting.OrderAccounting;
+import fi.hel.verkkokauppa.order.model.accounting.OrderItemAccounting;
+import fi.hel.verkkokauppa.order.repository.jpa.AccountingSlipRepository;
+import fi.hel.verkkokauppa.order.repository.jpa.AccountingSlipRowRepository;
+import fi.hel.verkkokauppa.order.service.order.OrderService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class AccountingSlipService {
+
+    private Logger log = LoggerFactory.getLogger(AccountingSlipService.class);
+
+    @Autowired
+    private OrderService orderService;
+
+    @Autowired
+    private OrderAccountingService orderAccountingService;
+
+    @Autowired
+    private OrderItemAccountingService orderItemAccountingService;
+
+    @Autowired
+    private AccountingSlipRepository accountingSlipRepository;
+
+    @Autowired
+    private AccountingSlipRowRepository accountingSlipRowRepository;
+
+    public AccountingSlip getAccountingSlip(String accountingSlipId) {
+        Optional<AccountingSlip> mapping = accountingSlipRepository.findById(accountingSlipId);
+
+        if (mapping.isPresent())
+            return mapping.get();
+
```

Comment:
this is the better place for producing NOT_FOUND exceptions, need to refactor the other apis later